apps/library/lib/library/game.ex

```diff
@@ -34,80 +35,104 @@ defmodule Library.Game do
     GenServer.cast(pid, {:remove_player, player})
   end
 
+  def display_state(pid), do: GenServer.call(pid, :display_state)
   def list_state(pid), do: GenServer.call(pid, :list_state)
 
   def init(:ok) do
     {
       :ok,
       %{
-        board: build_board,
+        board: Board.generate,
         players: [],
+        wordlist: [],
         next_index: 1
       }
     }
   end
 
-  def handle_call({:submit_word, word, player}, _from, board_state) do
+  def handle_call({:submit_word, word, player}, _from, game_state) do
     # this will delegate out to the dictionary to see if this is a valid word
     #
     # Would love to use this type of system.
     #
     # with {:ok} <- words_consist_of_valid_letters?(word),
     #      {:ok} <- have_not_played_word?(player, word),
-    # do:  {:reply, {:ok, word}, board_state}
-    # else {:reply, {:error, "word not valid", board_state}}
-
-    if true do
-      new_board = Enum.reduce(word, board_state.board, fn(letter, acc) ->
-        replace_letter(letter, player, acc) 
-      end)
+    # do:  {:reply, {:ok, word}, game_state}
+    # else {:reply, {:error, "word not valid", game_state}}
 
-      new_state = %{board_state | board: new_board}
+    # change submitted letters to use atoms for keys rather than strings
+    word = atomize_word(word)
+    new_board = Board.add_word(game_state.board, word, player)
+    new_state =
+      %{
+        game_state |
+          board: Board.surrounded(new_board),
+          wordlist: [word_from_letters(word)] ++ game_state.wordlist
```

Comment:
No, the wordlist is per game - you can't play a word anyone else has played previously. I haven't hooked up the logic for that yet.